src/pages/ServiceList/ServiceListComponent.tsx

```diff
@@ -1,67 +1,267 @@
+    this.pageSet = this.pageSet.bind(this);
+    this.pageSelect = this.pageSelect.bind(this);
+    this.updateSortField = this.updateSortField.bind(this);
+    this.updateSortDirection = this.updateSortDirection.bind(this);
+    this.state = {
+      loading: true,
+      services: [],
+      pagination: { page: 1, perPage: 10, perPageOptions: [5, 10, 15] },
+      currentSortField: sortFields[0],
+      isSortAscending: true
+    };
+  }
+
+  componentWillMount() {
+    this.updateServices();
+  }
+
+  filterChange() {
+    this.updateServices();
+  }
+
+  handleError(error: string) {
+    this.props.onError(error);
+    this.setState({ loading: false });
+  }
+
+  pageSet(page: number) {
+    this.setState(prevState => {
+      return {
+        loading: prevState.loading,
+        services: prevState.services,
+        pagination: {
+          page: page,
+          perPage: prevState.pagination.perPage,
+          perPageOptions: prevState.pagination.perPageOptions
+        }
+      };
+    });
+  }
+
+  pageSelect(perPage: number) {
+    this.setState(prevState => {
+      return {
+        loading: prevState.loading,
+        services: prevState.services,
+        pagination: {
+          page: 1,
+          perPage: perPage,
+          perPageOptions: prevState.pagination.perPageOptions
+        }
+      };
+    });
+  }
+
+  updateSortField(sortField: SortField) {
+    this.setState(prevState => {
+      return {
+        currentSortField: sortField,
+        services: this.sortServices(prevState.services, sortField, prevState.isSortAscending)
+      };
+    });
+  }
+
+  updateSortDirection() {
+    this.setState(prevState => {
+      return {
+        isSortAscending: !prevState.isSortAscending,
+        services: this.sortServices(prevState.services, prevState.currentSortField, !prevState.isSortAscending)
+      };
+    });
+  }
+
+  updateServices() {
+    const activeFilters: ActiveFilter[] = ServiceFilterSelected.getSelected();
+    let namespacesSelected: string[] = activeFilters
+      .filter(activeFilter => activeFilter.category === 'Namespace')
+      .map(activeFilter => activeFilter.value);
+    let servicenameFilters: string[] = activeFilters
+      .filter(activeFilter => activeFilter.category === 'Service Name')
+      .map(activeFilter => activeFilter.value);
+
+    if (namespacesSelected.length === 0) {
+      API.GetNamespaces()
+        .then(namespacesResponse => {
+          const namespaces: Namespace[] = namespacesResponse['data'];
+          this.fetchServices(namespaces.map(namespace => namespace.name), servicenameFilters);
+        })
+        .catch(namespacesError => {
+          console.error(JSON.stringify(namespacesError));
+          this.handleError('Error fetching namespace list.');
+        });
+    } else {
+      this.fetchServices(namespacesSelected, servicenameFilters);
+    }
+  }
+
+  fetchServices(namespaces: string[], servicenameFilters: string[]) {
```

Comment:
Or just: `const promises = namespaces.map(ns => API.GetServices(ns))`
(no blocker)

By the way: I've seen that API functions are Upper camel case and not lower, I started to change it in another PR, it's weird to see a mix of up and low camel case.

Reply:
> By the way: I've seen that API functions are Upper camel case and not lower, I started to change it in another PR, it's weird to see a mix of up and low camel case.

I thought it was some convention, I'm not against that, but as we are time constraint I would focus in consolidate before to introduce refactoring that are just code style.